src/items/external-blocks.md

```diff
@@ -121,19 +121,34 @@ specifies the kind of library with the following possible values:
 - `framework` — Indicates a macOS framework. This is only valid for macOS
   targets.
 
-The `name` key must be included if `kind` is specified.
+The `name` key must be included if `kind` is specified. Depending on the target, `name`
+is interpreted as follows:
+
+- `*-pc-windows-msvc` — Refers to the base name of the static library `name.lib`,
+or the base name of the import library `name.lib` accompanying a dynamic-link library `name.dll`.
+- All other targets — Refers to the base name of the shared library.
 
 The `wasm_import_module` key may be used to specify the [WebAssembly module]
 name for the items within an `extern` block when importing symbols from the
 host environment. The default module name is `env` if `wasm_import_module` is
 not specified.
 
 ```rust,ignore
+// Links to `libcrypto.so` on Linux, `libcrypto.dylib` on Mac, and the
+// (`crypto.lib`, `crypto.dll`) pair on windows-msvc.
```

Comment:
Rust isn't linking to `crypto.dll` at all on `windows-msvc`. The linker never looks for that file. Linking is strictly done through the `.lib` where some symbols resolve to a runtime dependency on a `.dll`.

`windows-gnu` is the odd one out as the MinGW linker is able to link against both import libraries and dlls directly so `-lfoo` looks for both `libfoo.a` and `foo.dll`. Of course if you're using the LLVM linker for `windows-gnu` then you're back to only linking against import libraries, so you're better off just always linking against the import library on windows except for the weird fact that Rust chooses not to emit import libraries for `cdylib`s on `windows-gnu` targets...

Reply:
Sure. What text would you propose instead?

Reply:
cc @retep998 -- any thought on ^--- (in general, you're the expert on this issue, and we, the reference maintainers are... not. ^^)

Reply:
Better phrasing might be "links against `crypto.lib` resulting in a runtime dependency on `crypto.dll`"

Reply:
Updated PR. Slightly changed your wording to fit into the sentence.